starboard/android/apk/app/src/main/java/dev/cobalt/coat/CobaltActivity.java

```diff
@@ -328,6 +328,16 @@ public void onRequestPermissionsResult(
     getStarboardBridge().onRequestPermissionsResult(requestCode, permissions, grantResults);
   }
 
+  public void clearVideoSurface() {
+    runOnUiThread(
```

Comment:
The clearing should be done very quick but still need a few time. After moving the clearing to the ui thread, it would no longer block player destroying. If user exit a playback and click on another playback very quick, the new player will need the SurfaceView, and would there be any problem? Like seeing the frame from previous video, cannot connect to the Surface as videoSurfaceView.clearSurface() still running?

Reply:
I implemented this to mirror `resetVideoSurface`, which is also called from `ClearVideoWindow` in video_window.cc.

If I instead call `videoSurfaceView.clearSurface();` directly, the application deadlocks when the playback stops.

Reply:
ClearVideoWindow() was running synchronously, which would block MianWebModule until it finished. The new code calls  `videoSurfaceView.clearSurface()` asynchronously. Theoretically, it's possible that a new player could be created before  `videoSurfaceView.clearSurface()` finishes. I tested it on Sabrina, after destroying the old player, it takes about ~1s to create a new player, so on sabrina, it seems to be fine. If I enforce a sleep of 1s between `Canvas canvas = getHolder().lockCanvas();` and `getHolder().setFormat(PixelFormat.OPAQUE);`, it would easily get a playback error (due to surface view "already connected"). It would be very risky if we merge the cl without extra protection.

Reply:
Thanks for the reproduction tip. I'll investigate.

Reply:
Ah, the dealock without the runnable is: The call the `clearVideoSurface` triggers a  `SurfaceHolderCallback::surfaceDestroyed`, which calls `nativeOnVideoSurfaceChanged`, which locks `lock(*GetViewSurfaceMutex())` which is already held inside `ClearVideoWindow` when it calls `clearVideoSurface`.

Given that the comment explaining the reason for holding the lock says it's for `g_native_video_window`, we can simply call `clearVideoSurface` after releasing the lock instead, and not need the runnable anymore in `CobaltActivity`.